weblate/vcs/git.py

```diff
@@ -279,6 +280,33 @@ def configure_remote(self, pull_url, push_url, branch):
         )
         self.branch = branch
 
+    def api_url(self):
+        return self.component.repo.replace(".git", "").replace(
+            "github.com", "api.github.com/repos"
+        )
```

Comment:
The repo URL can have several forms and can include HTTP authentication, the code should deal with that, here is what all we support on hooks side:

https://github.com/WeblateOrg/weblate/blob/6c44517f5330ddcdc19f6934c8f30e02c90f844a/weblate/trans/views/hooks.py#L58-L65